dotCMS/src/integration-test/java/com/dotcms/mock/request/CachedParameterDecoratorTest.java

```diff
@@ -0,0 +1,39 @@
+package com.dotcms.mock.request;
+
+import com.dotcms.IntegrationTestBase;
+import com.dotcms.datagen.LanguageDataGen;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class CachedParameterDecoratorTest extends IntegrationTestBase {
+
+
+    @BeforeClass
+    public static void prepare() throws Exception{
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    @Test
+    public void testDecorate() throws Exception {
+
+        final Language baseLanguage  = new LanguageDataGen().languageCode("it").countryCode(null).nextPersisted();
+        final Language itLanguage    = new LanguageDataGen().languageCode("it").countryCode("IT").nextPersisted();
```

Comment:
Use a randomly generated language and country code, otherwise, this can/will conflict with other data.